### ISSUE TITLE
Do we reuse pretrained parameter which learned the synthetic dataset?

### ISSUE THREAD

**Issue by hukim1112** [rpautrat/SuperPoint#37]
Hi!

Thank you for your repo.

I am wondering we reuse the checkpoint which learned synthetic shape dataset when we train the model with MS COCO(step 3 part of Usage in README file). I tried to find restoring part of that, but I can't.

When I read the paper, authors pretrain the model about synthetic dataset and retrain about MS coco dataset which have pseudo keypoint labels detected by the model pretrained by synthetic dataset.

Thank you.

---------------------------------------------------------------------------------------------------------------------------------------
In base_model.py

TensorFlow do global initialization in the last part of _build_graph.  

But there is no restoring parameter in train method.

    def train(self, iterations, validation_interval=100, output_dir=None, profile=False,
              save_interval=None, checkpoint_path=None, keep_checkpoints=1):
        assert self.trainable, 'Model is not trainable.'
        assert 'training' in self.datasets, 'Training dataset is required.'
        if output_dir is not None:
            train_writer = tf.summary.FileWriter(output_dir)
        if not hasattr(self, 'saver'):
            with tf.device('/cpu:0'):
                self.saver = tf.train.Saver(save_relative_paths=True,
                                            max_to_keep=keep_checkpoints)
        if not self.graph.finalized:
            self.graph.finalize()
        if profile:
            options = tf.RunOptions(trace_level=tf.RunOptions.FULL_TRACE)
            run_metadata = tf.RunMetadata()
        else:
            options, run_metadata = None, None

        tf.logging.info('Start training')
        for i in range(iterations):
            loss, summaries, _ = self.sess.run(
                    [self.loss, self.summaries, self.trainer],
                    feed_dict={self.handle: self.dataset_handles['training']},
                    options=options, run_metadata=run_metadata)

            if save_interval and checkpoint_path and (i+1) % save_interval == 0:
                self.save(checkpoint_path)
            if 'validation' in self.datasets and i % validation_interval == 0:
                metrics = self.evaluate('validation', mute=True)
                tf.logging.info(
                        'Iter {:4d}: loss {:.4f}'.format(i, loss) +
                        ''.join([', {} {:.4f}'.format(m, metrics[m]) for m in metrics]))

                if output_dir is not None:
                    train_writer.add_summary(summaries, i)
                    metrics_summaries = tf.Summary(value=[
                        tf.Summary.Value(tag=m, simple_value=v)
                        for m, v in metrics.items()])
                    train_writer.add_summary(metrics_summaries, i)

                    if profile and i != 0:
                        fetched_timeline = timeline.Timeline(run_metadata.step_stats)
                        chrome_trace = fetched_timeline.generate_chrome_trace_format()
                        with open(osp.join(output_dir,
                                           'profile_{}.json'.format(i)), 'w') as f:
                            f.write(chrome_trace)
        tf.logging.info('Training finished')

**Comment by rpautrat** [comment]
Hi!
There is indeed no possibility to restore previous weights when training for now. I was training from scratch each time and since the trainings are not so long, I found out that it was still fine like this and with good results.
But I agree that being able to restore weights would be convenient, I will see if I have time to add this option later.

**Comment by DongsuShin322** [comment]
Thank you for your response. It's clear now!

**Comment by hukim1112** [comment]
Thank you. It's clear now!
